server/hedley/modules/custom/hedley_stats/hedley_stats.module

```diff
@@ -1557,6 +1798,337 @@ function hedley_stats_add_immunization_dates_for_bundle(SelectQuery $base_query,
+
+  // Resolve all participants associated with those encounters.
+  hedley_general_join_field_to_query($query, 'node', 'field_individual_participant');
+  $query->groupBy('field_individual_participant.field_individual_participant_target_id');
+
+  $result = $query
+    ->execute()
+    ->fetchAllAssoc('field_individual_participant');
+
+  $individual_participants = array_keys($result);
+  if (empty($individual_participants)) {
+    $result = [];
+    hedley_stats_handle_cache(HEDLEY_STATS_CACHE_SET, HEDLEY_STATS_SYNC_NUTRITION_INDIVIDUAL_DATA, $health_center_id, NULL, $result);
+    return $result;
+  }
+
+  // Pull all participants to which resolved encounters belong, including
+  // some patient data.
+  $query = hedley_stats_get_base_query($health_center_id, 'individual_participant');
+  $query->condition('node.nid', $individual_participants, 'IN');
+
+  $fields_to_join = [
+    'field_person',
+  ];
+  foreach ($fields_to_join as $field) {
+    hedley_general_join_field_to_query($query, 'node', $field, FALSE);
+  }
+
+  // Add patient birthdate, so we can calculate the age.
+  $query->leftJoin('field_data_field_birth_date', 'birth_date', 'birth_date.entity_id = field_person.field_person_target_id');
+  $query->addField('birth_date', 'field_birth_date_value');
+
+  $result = hedley_stats_run_node_query_in_batches($query);
+
+  $items = [];
+  // Prepare participant data to be consumed by client side.
+  foreach ($result as $row) {
+    $items[$row->nid] = [
+      'id' => $row->field_person,
+      // Field for 'created' is added at hedley_stats_get_base_query().
+      'created' => date_format(date_create($row->field_expected), 'Y-m-d'),
+      'birth_date' => date_format(date_create($row->field_birth_date_value), 'Y-m-d'),
+      'encounters' => [],
+    ];
+  }
+
+  // Free up memory.
+  drupal_static_reset();
+
+  // Resolve encounters data.
+  $base_query = hedley_stats_get_base_query($health_center_id, 'nutrition_encounter');
+  $fields_to_join = [
+    'field_individual_participant',
+    'field_nutrition_encounter_type',
+  ];
+  foreach ($fields_to_join as $field) {
+    hedley_general_join_field_to_query($base_query, 'node', $field, FALSE);
+  }
+  $base_query->condition('field_individual_participant.field_individual_participant_target_id', $individual_participants, 'IN');
+
+  // Join with Nutrition Encounter table, through which we can
+  // join measurements data.
+  $base_query->leftJoin('field_data_field_nutrition_encounter', 'nutrition_encounter', 'nutrition_encounter.field_nutrition_encounter_target_id = node.nid');
+
+  $query = clone $base_query;
+
+  // Pull data of MUAC that was recorded during encounter.
+  $query->leftJoin('field_data_field_muac', 'muac', 'muac.entity_id = nutrition_encounter.entity_id');
+  $query->addField('muac', 'field_muac_value');
+  $query->addExpression("GROUP_CONCAT(DISTINCT muac.field_muac_value)", 'muac');
```

Comment:
Don't we need all MUAC values here? With DISTINCT, we may drop some of the values.

Reply:
Well, actually, here we add data of MUAC measurement to individual encounter measurements. There can be only single MUAC measurement, so we should be safe with DISTINCT.